api/handlers_spaces.go

```diff
@@ -37,15 +48,68 @@ func (s *server) SpaceGetHandler(w http.ResponseWriter, r *http.Request) {
 	}
 	log.Debugf("found cost explorer result cache %+v", *resultCache)
 
-	spaceID := vars["space"]
-	log.Debugf("getting costs for space %s", spaceID)
+	// vars for checking input times parse and are valid
+	timeValidity := false
+	var scleansedTime string
+	var ecleansedTime string
+	// Did we get cost-explorer start and end times on the API?
+	// set defaults, else verify times given on API
+	if endTime == "" || startTime == "" {
+		log.Debug("no start or end time given on API input, assigning defaults")
+		// if it's the first day of the month, get today's usage thus far
+		y, m, d := time.Now().Date()
+		if d == 1 {
+			d = 2
+		}
 
-	y, m, d := time.Now().Date()
+		ecleansedTime = fmt.Sprintf("%d-%02d-%02d", y, m, d)
+		scleansedTime = fmt.Sprintf("%d-%02d-01", y, m)
+		timeValidity = true
+	} else {
+		// working vars for time verification
+		var sTime string
+		var eTime string
+		var badTime bool
+
+		// debugging verbosity
+		log.Debugf("startTime: %s", startTime)
+		log.Debugf("endTime: %s", endTime)
 
-	// if it's the first day of the month, get todays usage thus far
-	if d == 1 {
-		d = 2
+		// set the time parse layout and do parse
+		layoutISO := "2006-01-02"
+		start, err := time.Parse(layoutISO, startTime)
+		if err != nil {
+			msg := fmt.Sprintf("error parsing startTime from input: %s\n", err)
+			handleError(w, apierror.New(apierror.ErrBadRequest, msg, nil))
+			badTime = true
+		}
+		end, err := time.Parse(layoutISO, endTime)
+		if err != nil {
+			msg := fmt.Sprintf("error parsing endTime from input: %s\n", err)
+			handleError(w, apierror.New(apierror.ErrBadRequest, msg, nil))
+			badTime = true
+		}
+
+		// convert times to number string yyMMdd so we can do a math comparison
+		sTime = fmt.Sprint(start.Format("20060102"))
+		eTime = fmt.Sprint(end.Format("20060102"))
+		log.Debugf("startYYYYMMDD: %s", sTime)
+		log.Debugf("endYYYYMMDD: %s", eTime)
```

Comment:
you can get rid of all of this (and below) and use https://golang.org/pkg/time/#Time.After

Reply:
nice - timeValidity = eTmp.After(sTmp)